apps/server/src/modules/deletion/repo/deletion-log.repo.spec.ts

```diff
@@ -0,0 +1,186 @@
+import { EntityManager, ObjectId } from '@mikro-orm/mongodb';
+import { Test } from '@nestjs/testing';
+import { TestingModule } from '@nestjs/testing/testing-module';
+import { MongoMemoryDatabaseModule } from '@shared/infra/database';
+import { cleanupCollections } from '@shared/testing';
+import { DeletionLogMapper } from './mapper';
+import { DeletionLogEntity } from '../entity';
+import { DeletionLogRepo } from './deletion-log.repo';
+import { deletionLogFactory } from '../domain/testing/factory/deletion-log.factory';
+import { DeletionLog } from '../domain/deletion-log.do';
+import { deletionLogEntityFactory } from '../entity/testing/factory/deletion-log.entity.factory';
+
+describe(DeletionLogRepo.name, () => {
+	let module: TestingModule;
+	let repo: DeletionLogRepo;
+	let em: EntityManager;
+
+	beforeAll(async () => {
+		module = await Test.createTestingModule({
+			imports: [
+				MongoMemoryDatabaseModule.forRoot({
+					entities: [DeletionLogEntity],
+				}),
+			],
+			providers: [DeletionLogRepo, DeletionLogMapper],
+		}).compile();
+
+		repo = module.get(DeletionLogRepo);
+		em = module.get(EntityManager);
+	});
+
+	afterAll(async () => {
+		await module.close();
+	});
+
+	afterEach(async () => {
+		await cleanupCollections(em);
+	});
+
+	describe('defined', () => {
+		it('repo should be defined', () => {
+			expect(repo).toBeDefined();
+		});
+
+		it('entity manager should be defined', () => {
+			expect(em).toBeDefined();
+		});
+
+		it('should implement entityName getter', () => {
+			expect(repo.entityName).toBe(DeletionLogEntity);
+		});
+	});
+
+	describe('create deletionLog', () => {
+		describe('when deletionLog is new', () => {
+			it('should create a new deletionLog', async () => {
+				const domainObject: DeletionLog = deletionLogFactory.build();
+				const deletionLogId = domainObject.id;
+				await repo.create(domainObject);
+
+				const expectedDomainObject = {
+					id: domainObject.id,
+					domain: domainObject.domain,
+					operation: domainObject.operation,
+					modifiedCount: domainObject.modifiedCount,
+					deletedCount: domainObject.deletedCount,
+					deletionRequestId: domainObject.deletionRequestId,
+					createdAt: domainObject.createdAt,
+					updatedAt: domainObject.updatedAt,
+				};
```

Comment:
Please move these to the `setup()` function.

Reply:
done